lambdas/mail_send.js

```diff
@@ -4,17 +4,47 @@ if (process.env.NODE_ENV === 'development') {
 
 const domain = process.env.MAILGUN_DOMAIN;
 const apiKey = process.env.MAILGUN_API_KEY;
+const secret_key = process.env.SECRET_SEND_KEY;
 
+const { send } = require('micro');
+const statuses = require('statuses');
+const { upload } = require('micro-upload')
 const mailgun = require('mailgun-js');
 
-const mg = mailgun({ apiKey, domain });
-const data = {
-  from: 'WebConf <no-reply@webconf.tech>',
-  to: 'gahs94@gmail.com',
-  subject: 'WebConf • Nuestro newsletter 🙈',
-  text: 'Hola perrix! :D'
-};
+module.exports = upload(async (req, res) => {
+  if (!(req.files && req.files.file && req.files.file.mimetype === 'text/html')) {
+    return send(res, statuses['bad request'], 'Error - No file uploaded or wrong file format');
+  }
 
-module.exports = (req, res) => {
-  mg.messages().send(data, (error, body) => console.log(body));
-};
+  const { listName, subject, specificAddress, secret } = req.body;
+  if (secret === secret_key) {
+    const messageData = {
+      subject,
+      to: specificAddress,
+      html: req.files.file.data.toString('utf8'),
+      from: 'WebConf <no-reply@webconf.tech>'
+    };
+    if (listName) {
+      try {
+        const listMembers = await mailgun({ domain, apiKey }).lists(listName).members().list();
+        const subscribedMembers = listMembers.items.filter(user => user.subscribed);
+
+        messageData.to = subscribedMembers.map(user => user.address);
+        messageData['recipient-variables'] = subscribedMembers.reduce(
```

Comment:
This probably goes against some functional programming concept (cc @asermax), but it's my opinion:

Instead of using both `.map` and `.reduce`, use `.forEach` to push the objects and iterate the array only once:

```js
messageData.to = [];
messageData['recipient-variables'] = {};
subscribedMembers.forEach((user) => {
  messageData.to.push(user.address);
  messageData['recipient-variables'][user.address] = user;
});
```

Reply:
`forEach` is not pure, but we're not writing purely functional cod so it's okay :stuck_out_tongue: 
In defense of  @spersico, the amount of data being managed is so little that his approach should be fine performance wise (and I like it more the way he did it).

Reply:
I'd stick with the current implementation. `forEach` with side effects makes a bit of noise for me too.

Reply:
@asermax Yes, for this, this kind of change would go as an unnecessary micro optimization; But I asked him, "how annoying can I be on the review?", he asked for this xD.

@joelalejandro If you were talking about `map`, `reduce`, `filter` or any of those, I would totally agree with you; but `forEach` just iterates the array, I can only think of two or three cases where you won't be creating side effects.